scripts/remoteperformance.ps1

```diff
@@ -0,0 +1,296 @@
+}
+
+$CurrentCommitHash = Get-GitHash -RepoDir $RootDir
+
+function LocalSetup {
+    $RetObj = New-Object -TypeName psobject
+    $RetObj | Add-Member -MemberType NoteProperty -Name apipaInterfaces -Value $null
+    try {
+        if ($IsWindows) {
+            $apipaAddr = Get-NetIPAddress 169.254.*
+            if ($null -ne $apipaAddr) {
+                # Disable all the APIPA interfaces for URO perf.
+                Write-Debug "Temporarily disabling APIPA interfaces"
+                $RetObj.apipaInterfaces = (Get-NetAdapter -InterfaceIndex $apipaAddr.InterfaceIndex) | Where-Object {$_.AdminStatus -eq "Up"}
+                $RetObj.apipaInterfaces | Disable-NetAdapter -Confirm:$false
+            }
+        }
+    } catch {
+        $RetObj.apipaInterfaces = $null
+    }
+
+    return $RetObj
+}
+
+function LocalTeardown {
+    param ($LocalCache)
+    if ($null -ne $LocalCache.apipaInterfaces) {
+        # Re-enable the interfaces we disabled earlier.
+        Write-Debug "Re-enabling APIPA interfaces"
+        $LocalCache.apipaInterfaces | Enable-NetAdapter
+    }
+}
+
+function Invoke-Test {
+    param ($Test)
+
+    Write-Output "Running Test $Test"
+
+    $RemoteExe = Get-ExeName -PathRoot $RemoteDirectory -Platform $RemotePlatform -IsRemote $true -TestPlat $Test.Remote
+    $LocalExe = Get-ExeName -PathRoot $LocalDirectory -Platform $LocalPlatform -IsRemote $false -TestPlat $Test.Local
+
+    # Check both Exes
+    $RemoteExeExists = Invoke-TestCommand -Session $Session -ScriptBlock {
+        param ($RemoteExe)
+        Test-Path $RemoteExe
+    } -ArgumentList $RemoteExe
+
+    $LocalExeExists = Test-Path $LocalExe
+
+    if (!$RemoteExeExists -or !$LocalExeExists) {
+
+        if (!$RemoteExeExists) {
+            Write-Output "Missing Remote Exe $RemoteExe"
+        }
+        if (!$LocalExeExists) {
+            Write-Output "Missing Local Exe $LocalExe"
+        }
+        Write-Error "Failed to Run $Test because of missing exe"
+    }
+
+    $LocalArguments = $Test.Local.Arguments.GetArguments().Replace('$RemoteAddress', $RemoteAddress)
+    $LocalArguments = $LocalArguments.Replace('$LocalAddress', $LocalAddress)
+
+    $CertThumbprint = Invoke-TestCommand -Session $Session -ScriptBlock {
+        return $env:QUICCERT
+    }
+
+    $RemoteArguments = $Test.Remote.Arguments.GetArguments().Replace('$Thumbprint', $CertThumbprint)
+
+    $RemoteJob = Invoke-RemoteExe -Exe $RemoteExe -RunArgs $RemoteArguments
+
+    $ReadyToStart = Wait-ForRemoteReady -Job $RemoteJob
+
+    if (!$ReadyToStart) {
+
+        Stop-Job -Job $RemoteJob
+        Write-Error "Test Remote for $Test failed to start"
+    }
+
+    $AllRunsResults = @()
+
+    try {
+        1..$Test.Iterations | ForEach-Object {
+            $LocalResults = Invoke-LocalExe -Exe $LocalExe -RunArgs $LocalArguments
+
+            $LocalParsedResults = Get-TestResult -Results $LocalResults -Matcher $Test.ResultsMatcher
+
+            $AllRunsResults += $LocalParsedResults
```

Comment:
nit: you don't need all these extra newlines between things. In general, don't put newlines between single line code blocks.